specifyweb/specify/views.py

```diff
@@ -391,3 +392,84 @@ def set_admin_status(request, userid):
     else:
         user.clear_admin()
         return http.HttpResponse('false', content_type='text/plain')
+
+@openapi(schema={
+    'post': {
+        "requestBody": {
+            "required": True,
+            "description": "Replace a new agent for an old agent.",
+            "content": {
+                "application/x-www-form-urlencoded": {
+                    "schema": {
+                        "type": "object",
+                        "description": "The error.",
+                        "properties": {
+                            "old_agent_id": {
+                                "type": "integer",
+                                "description": "The old AgentID value of the agent record that is to be replaced by the new one."
+                            },
+                            "new_agent_id": {
+                                "type": "integer",
+                                "description": "The new AgentID value of the agent that is replacing the old one."
+                            }
+                        },
+                        'required': ['old_agent_id', 'new_agent_id'],
+                        'additionalProperties': False
+                    }
+                }
+            }
+        },
+        "responses": {
+            "204": {"description": "Success",},
+            "400": {"description": "The AgentID specified does not exist."}
+        }
+    },
+})
+@login_maybe_required
+@require_POST
+def agent_record_replacement(request: http.HttpRequest, old_agent_id, new_agent_id: int) -> http.HttpResponse:
+    """Replaces all the foreign keys referencing the old AgentID
+    with the new AgentID, and deletes the old agent record.
+    """
+    check_permission_targets(request.specify_collection.id, request.specify_user.id, [])
+
+    # Create database connection cursor
+    cursor = connection.cursor()
+
+    with transaction.atomic():
+        # Check to make sure both the old and new agent IDs exist in the table
+        cursor.execute("SELECT * FROM specify.agent WHERE AgentID = %s;", [old_agent_id])
```

Comment:
I would make these SELECT FOR UPDATE statements. In the default isolation level another transaction can delete the rows after you have selected them. So in a race condition you could find the agents exist here, but they are gone by the time you try to use them. FOR UPDATE locks the rows until your transaction completes.

I would also consider using the Django ORM for these checks. It could make the intent a little clearer. Something like 
```
if not models.Agent.objects.filter(id=old_agent_id).select_for_update().exists():
   ...
```

I think `select_for_update` and `exists` can be combined in that manner. It's a matter of taste though, and you should use whichever you prefer since the function is already using raw SQL.